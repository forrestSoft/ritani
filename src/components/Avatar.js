import React from 'react';

function Avatar(props){
	return(
		<React.Fragment>
			<img src={props.url} />
		</React.Fragment>
	)
}

export default Avatar;