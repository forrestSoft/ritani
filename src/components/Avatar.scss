.avatar_wrapper {
	position: relative;
  display: flex;
  margin: 3px;  
  background:#fff;
  padding: 5px 5px 16px 5px;

  input {
  	display: block;
  	margin-top: 2vh;
    font-size: calc(3vh/2); 
  }

  img {
    height: 50px; 
    width: 50px;
    z-index: 2;
    border: 1px solid #ccc;
    background-clip: content-box;
    box-sizing: border-box;
  }
  
  @keyframes img {
    to {transform: rotate(360deg);}
  }
   
  &:before {
    content: '';
    box-sizing: border-box;
    position: absolute;
    top: 25%;
    left: 28%;
    width: 40%;
    height: 30%;
    border-radius: 50%;
    border: 1px solid #ccc;
    border-top-color: #07d;
    animation: img .6s linear infinite;
    box-sizing: border-box;
  }
}