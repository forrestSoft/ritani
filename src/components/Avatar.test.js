import React from 'react'
import {shallow} from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import Avatar from './Avatar'

// shallow render and check props are working
it('render matches snapshot', () => {
	const url = "https://avatars2.githubusercontent.com/u/1555350?v=4"
  const avatar = shallowToJson(shallow(<Avatar url={url} />))
  expect(avatar).toMatchSnapshot();
})
