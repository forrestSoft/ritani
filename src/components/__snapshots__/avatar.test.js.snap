// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<Fragment>
  <img
    src="https://avatars2.githubusercontent.com/u/1555350?v=4"
  />
</Fragment>
`;
